### PR TITLE
add charset UTF-8 to log api

### PR DESCRIPTION
**Pull request by dyf6372**
When I retrieve the task log in browser, the Chinese characters all end up as garbage.
image
After adding charset UTF-8, it was correct.
image

### CODE REVIEW

**Comment by asdf2014** [comment]
Hi, @dyf6372 . Thanks for your contribution. This problem may appear in all places where the `@Produces` annotation is used. FYI, we'd better fix it at once by adding a `ContainerRequestFilter` subclass instead of modifying the `@Produces` annotation everywhere.

**Comment by leventov** [comment]
@dyf6372 please add a Checkstyle Regex rule that prohibits adding `@Produces("text/plain")` in the future

**Comment by jihoonson** [comment]
@asdf2014 @leventov thanks for the review. I think it makes sense to raise a follow-up pr to apply this to everywhere else. What do you think?

**Comment by asdf2014** [comment]
Hi, @jihoonson. Right, this solution is the most simplified. It would be better to use `MediaType.TEXT_PLAIN +"; charset = UTF-8"` instead of `"text / plain"`, which can be declared as a TEXT_PLAIN_UTF8 constant.

> can not fix it by adding a ContainerResponseFilter since there are no related APIs we can use to do

Hi, @QiuMM . Thanks for your comment. I tried program a `MediaTypeFilter` demo, but it's not be tested.

image

image


> And for @Produces(MediaType.APPLICATION_JSON), I had impression that the default encoding for JSON is UTF-8 

For this point, I cannot agree with you :sweat_smile:

**Comment by dyf6372** [comment]
In [rfc4627](https://tools.ietf.org/html/rfc4627) section 3, the default encoding of JSON text is UTF-8.
_JSON text SHALL be encoded in Unicode.  The default encoding is UTF-8._ 
_Since the first two characters of a JSON text will always be ASCII characters [RFC0020], it is possible to determine whether an octet stream is UTF-8, UTF-16 (BE or LE), or UTF-32 (BE or LE) by looking at the pattern of nulls in the first four octets._
Since it is possible to determine whether an octet stream is UTF-8, UTF-16, or UTF-32, I'm confused whether to set the charset to UTF-8 for MediaType.APPLICATION_JSON.  Maybe is same to MediaType.TEXT_PLAIN.

**Comment by asdf2014** [comment]
Hi, @dyf6372 . Yep, the encoding of JSON should be UTF-8, which is actually an industry standard. However, this is not for Jersey :sweat_smile:

**Comment by dyf6372** [comment]
@asdf2014 just like what your said, I declare a constant named TEXT_PLAIN_UTF8 instead of `"text/plain; charset=UTF-8"`, and add a Checkstyle Regex rule that prohibits adding `@Produces("text/plain")`. 
@asdf2014  @leventov Please help me review whether the new change is's ok? If it is fine, I will raise a follow-up pr to apply this to JSON.

File: codestyle/checkstyle.xml [hunk 1]
```diff
@@ -194,6 +194,13 @@
       <property name="message" value="Use org.apache.druid.common.config.NullHandling.nullToEmptyIfNeeded instead"/>
     </module>
 
+
```
[the remark this answers is not in the document]
Comment (dyf6372) [reply codestyle/checkstyle.xml]: Good, I'll remove it.

File: codestyle/checkstyle.xml [hunk 2]
```diff
@@ -194,6 +194,13 @@
...
+    <module name="Regexp">
+      <property name="format" value="@Produces\(&quot;text/plain&quot;\)"/>
+      <property name="illegalPattern" value="true"/>
+      <property name="message" value="Don't forget the charset"/>
```
[the remark this answers is not in the document]
Comment (dyf6372) [reply codestyle/checkstyle.xml]: Good, I'll modify it.

File: codestyle/checkstyle.xml [hunk 3]
```diff
@@ -194,6 +194,13 @@
       <property name="message" value="Use org.apache.druid.common.config.NullHandling.nullToEmptyIfNeeded instead"/>
     </module>
 
+
+    <module name="Regexp">
+      <property name="format" value="@Produces\(&quot;text/plain&quot;\)"/>
```
[the remark this answers is not in the document]
Comment (dyf6372) [reply codestyle/checkstyle.xml]: I tried it before, but it was failed to use `\"` in xml.

[the remark this answers is not in the document]
Comment (asdf2014) [reply codestyle/checkstyle.xml]: Right, please also correct the line [176](https://github.com/apache/incubator-druid/blob/caf13e104f979af0dad22dac83069dbfca3597a2/codestyle/checkstyle.xml#L176).

File: codestyle/checkstyle.xml [hunk 4]
```diff
@@ -194,6 +194,12 @@
       <property name="message" value="Use org.apache.druid.common.config.NullHandling.nullToEmptyIfNeeded instead"/>
     </module>
 
+    <module name="Regexp">
+      <property name="format" value="@Produces\(&quot;text/plain&quot;\)"/>
+      <property name="illegalPattern" value="true"/>
+      <property name="message" value="Use org.apache.druid.server.http.HttpMediaType#TEXT_PLAIN_UTF8 instead would be better."/>
```
Comment (asdf2014) [inline codestyle/checkstyle.xml]: I mean `<property name="message" value="Use org.apache.druid.server.http.HttpMediaType#TEXT_PLAIN_UTF8 instead"/>`

**Pull request merged**